### PR TITLE
python311Packages.devito: 4.8.3 -> 4.8.6

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/devitocodes/devito/releases/tag/v4.8.6

cc @AtilaSaraiva 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Review by natsukium — COMMENTED** [review]
I guess we should mark broken for darwin and aarch64-linux?

File: pkgs/development/python-modules/devito/default.nix [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 buildPythonPackage rec {
   pname = "devito";
-  version = "4.8.3";
+  version = "4.8.4";
   format = "setuptools";
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/development/python-modules/devito/default.nix:30]: Unfortunately, it fails with the following (even after adding `setuptools` to `nativeBuildInputs`):
```
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for devito-4.8.4-py3-none-any.whl
  + Exception Group Traceback (most recent call last):
  |   File "/nix/store/gdxd1rn8zgjjw08ijghvpazj18n4sjl6-python-runtime-deps-check-hook.py", line 93, in <module>
  |     metadata = get_metadata(args.wheel)
  |                ^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/gdxd1rn8zgjjw08ijghvpazj18n4sjl6-python-runtime-deps-check-hook.py", line 58, in get_metadata
  |     metadata = Metadata.from_raw(raw)
  |                ^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/vy23p29z57xd9r7lmpdqhdbqv8rkvw7n-python3.11-packaging-24.0/lib/python3.11/site-packages/packaging/metadata.py", line 720, in from_raw
  |     raise ExceptionGroup("invalid metadata", exceptions)
  | ExceptionGroup: invalid metadata (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/nix/store/vy23p29z57xd9r7lmpdqhdbqv8rkvw7n-python3.11-packaging-24.0/lib/python3.11/site-packages/packaging/metadata.py", line 626, in _process_provides_extra
    |     normalized_names.append(utils.canonicalize_name(name, validate=True))
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/nix/store/vy23p29z57xd9r7lmpdqhdbqv8rkvw7n-python3.11-packaging-24.0/lib/python3.11/site-packages/packaging/utils.py", line 45, in canonicalize_name
    |     raise InvalidName(f"name is invalid: {name!r}")
    | packaging.utils.InvalidName: name is invalid: 'python_version_3.8_'
    | 
    | The above exception was the direct cause of the following exception:
    | 
    | Traceback (most recent call last):
    |   File "/nix/store/vy23p29z57xd9r7lmpdqhdbqv8rkvw7n-python3.11-packaging-24.0/lib/python3.11/site-packages/packaging/metadata.py", line 715, in from_raw
    |     getattr(ins, key)
    |   File "/nix/store/vy23p29z57xd9r7lmpdqhdbqv8rkvw7n-python3.11-packaging-24.0/lib/python3.11/site-packages/packaging/metadata.py", line 523, in __get__
    |     value = converter(value)
    |             ^^^^^^^^^^^^^^^^
    |   File "/nix/store/vy23p29z57xd9r7lmpdqhdbqv8rkvw7n-python3.11-packaging-24.0/lib/python3.11/site-packages/packaging/metadata.py", line 628, in _process_provides_extra
    |     raise self._invalid_metadata(
    | packaging.metadata.InvalidMetadata: 'python_version_3.8_' is invalid for 'provides-extra'
    +------------------------------------
```

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
  </ul>
</details>

**Comment by drupol** [comment]
Looks like there's a hash mismatch in the CI.

**Comment by GaetanLepage** [comment]
> Looks like there's a hash mismatch in the CI.

Fixed.

File: pkgs/development/python-modules/devito/default.nix [hunk 2]
```diff
@@ -121,5 +122,13 @@ buildPythonPackage rec {
     changelog = "https://github.com/devitocodes/devito/releases/tag/v${version}";
     license = licenses.mit;
     maintainers = with maintainers; [ atila ];
+    broken =
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/development/python-modules/devito/default.nix]: This was actually a problem with `contexttimer` which seems unmaintained and is not building on python 3.12.
This was required by devito's optional dependency pyevolve which I have thus removed.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 305047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]

**Pull request merged**